### PR TITLE
Fix loadBundle progress

### PR DESCRIPTION
**Pull request by Zyie**
`loadBundle`'s `onProgress` event would fire for the individual assets per bundle, resulting in something like this when loading multiple bundles

`0.3, 0.6, 1, 0.5, 1`

~~This fix only counts when all the bundle loads and produces the expected output~~

~~`0.5, 1`~~

This fix now uses the total assets of all bundles. If 2 bundles are loaded, each containing 2 assets it produces this

`0.25, 0.5, 0.75, 1`

### CODE REVIEW

**Comment by bigtimebuddy** [comment]
Wouldn't we want something like this instead?

0.15, 0.3, 0.5, 0.75, 1

onProgress for the bundles completed seems less helpful.

**Comment by Zyie** [comment]
> Wouldn't we want something like this instead?
> 
> 0.15, 0.3, 0.5, 0.75, 1
> 
> onProgress for the bundles completed seems less helpful.

That is a good point 😄 updating it now

**Comment by Zyie** [comment]
@bigtimebuddy updated

File: packages/assets/src/Assets.ts [hunk 1]
```diff
@@ -523,11 +523,20 @@ export class AssetsClass
 
         const out: Record<string, Record<string, any>> = {};
 
-        const promises = Object.keys(resolveResults).map((bundleId) =>
+        const keys = Object.keys(resolveResults);
+        let count = 0;
+        let total = 0;
+        const _onProgress = () =>
+        {
+            if (onProgress) onProgress(++count / total);
```
Comment (bigtimebuddy) [inline packages/assets/src/Assets.ts:531]: What about this?

```suggestion
            onProgress?.(++count / total);
```

**Review by bigtimebuddy — COMMENTED** [review]
What do you think about onProgress calling an initial value of 0? Could be helpful for folks animating a progress bar.

**Review by bigtimebuddy — APPROVED** [review]

**Review by GoodBoyDigital — APPROVED** [review]

**Pull request merged**